rasa/core/featurizers/intent_max_history_tracker_featurizer.py

```diff
@@ -0,0 +1,304 @@
+from typing import List, Text, Tuple, Dict, Any, Iterator, Optional
+from collections import defaultdict
+from tqdm import tqdm
+import numpy as np
+import logging
+
+from rasa.core.featurizers.tracker_featurizers import (
+    MaxHistoryTrackerFeaturizer,
+    is_prev_action_listen_in_state,
+)
+from rasa.shared.core.domain import State, Domain
+from rasa.shared.core.trackers import DialogueStateTracker
+from rasa.utils.tensorflow.constants import LABEL_PAD_ID
+from rasa.shared.core.events import ActionExecuted, UserUttered
+import rasa.shared.utils.io
+
+logger = logging.getLogger(__name__)
+
+
+class IntentMaxHistoryTrackerFeaturizer(MaxHistoryTrackerFeaturizer):
+    """Truncates the tracker history into `max_history` long sequences.
+
+    Creates training data from trackers where intents are the output prediction
+    labels. Tracker state sequences which represent policy input are truncated
+    to not excede `max_history` states.
+    """
+
+    LABEL_NAME = "intent"
+
+    @classmethod
+    def _convert_labels_to_ids(
+        cls, trackers_as_intents: List[List[Text]], domain: Domain
+    ) -> np.ndarray:
+        """Converts a list of labels to a matrix of label ids.
+
+        The number of rows is equal to `len(trackers_as_intents)`. The number of
+        columns is equal to the maximum number of positive labels that any training
+        example is associated with. Rows are padded with `LABEL_PAD_ID` if not all rows
+        have the same number of labels.
+
+        Args:
+            trackers_as_intents: Positive example label ids
+                associated with each training example.
+            domain: The domain of the training data.
+
+        Returns:
+           A matrix of label ids.
+        """
+        # store labels in numpy arrays so that it corresponds to np arrays
+        # of input features
+        label_ids = [
+            [domain.intents.index(intent) for intent in tracker_intents]
+            for tracker_intents in trackers_as_intents
+        ]
+
+        return np.array(cls._pad_label_ids(label_ids))
+
+    @staticmethod
+    def _pad_label_ids(label_ids: List[List[int]]) -> List[List[int]]:
+        """Pads label ids so that all are of the same length.
+
+        The pad value `LABEL_PAD_ID` is set in `rasa.utils.tensorflow.constants`.
```

Comment:
I think this statement is not needed in the docstring as it's evident from the import statement where it's defined and users can navigate. Moreover, if it gets refactored in the future to be placed somewhere else then I am sure we'll forget to modify this docstring 🙂